### ISSUE TITLE
Ability to overtype angle brackets (similar to normal parentheses)

### ISSUE THREAD

**Issue by fwcd** [microsoft/vscode#187720]
When using a language that auto-inserts angle brackets in code completions (such as Rust with `rust-analyzer` in generic type signatures) it would be very convenient if angle brackets could be "overtyped", similar to normal parentheses, as e.g. in method completions. A quick demo of the issue below:

https://github.com/microsoft/vscode/assets/30873659/7c134bf6-2e92-411a-8518-40e6b4a84147

**Comment by RedCMD** [comment]
so instead of inserting another `>` it should instead just move the cursor to outside the `<>`?
same as normal brackets

**Comment by fwcd** [comment]
Yes, exactly. Perhaps VSCode could detect if the angle-brackets were auto-inserted by a completion/snippet and otherwise keep the current behavior (of just inserting the character).

**Comment by hediet** [comment]
I thought languages can configure this?

**Comment by RedCMD** [comment]
extensions can configure it 
but Rust does not.
I assume cause it conflicts with less than `<` and less than or equal to `<=` operators too much

otherwise the `overtype` bracket feature does work perfectly fine

**Comment by fwcd** [comment]
@RedCMD Do you have an example for an extension that does this or how this could be done? (I haven't seen it, so I'd be curious)

Many languages, including HTML or TypeScript get around this issue by simply not auto-inserting the closing angle bracket.
